src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/repositories/ResourceContainerRepository.kt

```diff
@@ -0,0 +1,154 @@
+                val dublinCoreFk = resourceMetadataDao.insert(ResourceMetadataMapper().mapToEntity(metadata), dsl)
+
+                // TODO: Should this only happen if we are importing a "help" RC? What if the help is imported first,
+                // TODO: ... then another related bundle RC is imported?
+                val relatedDublinCoreIds: List<Int> =
+                        linkRelatedResourceContainers(dublinCoreFk, rc.manifest.dublinCore.relation, dsl)
+
+                // TODO: Probably shouldn't hardcode "help"
+                if (rc.type() == "help") {
+                    relatedDublinCoreIds.forEach { relatedId ->
+                        val ih = ImportHelper(dublinCoreFk, relatedId, dsl)
+                        ih.importCollection(null, rcTree)
+                    }
+                } else {
+                    val ih = ImportHelper(dublinCoreFk, null, dsl)
+                    ih.importCollection(null, rcTree)
+                }
+            }
+        }.subscribeOn(Schedulers.io())
+    }
+
+    private fun linkRelatedResourceContainers(
+            dublinCoreFk: Int,
+            relations: List<String>,
+            dsl: DSLContext)
+            : List<Int> {
+        val relatedIds = mutableListOf<Int>()
+        relations.forEach { relation ->
+            val parts = relation.split('/')
+            resourceMetadataDao.fetchByLanguageAndIdentifier(parts[0], parts[1], dsl)
+                    ?.let { relatedDublinCore ->
+                        // TODO: Only add link if it doesn't exist already
+                        resourceMetadataDao.addLink(dublinCoreFk, relatedDublinCore.id, dsl)
+                        relatedIds.add(relatedDublinCore.id)
+                    }
+        }
+        return relatedIds
+    }
+
+    inner class ImportHelper(val dublinCoreId: Int, val relatedBundleDublinCoreId: Int?, val dsl: DSLContext, var collectionsImported: Int = 0) {
+
+        fun importCollection(parentId: Int?, node: Tree) {
+            (node.value as? Collection)?.let { collection ->
+                when (relatedBundleDublinCoreId) {
+                    null -> addCollection(collection, parentId)
+                    else -> findCollectionId(collection, relatedBundleDublinCoreId)
+                }
+            }.let { collectionId ->
+                // TODO: If we don't find a corresponding collection, we continue on, passing null to collectionId.
+                // TODO ... Eventually, contents will not be created if there is no parentId. This will happen for front
+                // TODO ... matter until we have another solution.
+                for (childNode in node.children) {
+                    importNode(collectionId, childNode)
+                }
+            }
+            collectionsImported++
+            if (collectionsImported % 25 == 0) {
+                println("Collections imported: $collectionsImported")
```

Comment:
unsure if we should be committing print statements like this

Reply:
Yeah that wasn't supposed to make it into the PR, sorry

Reply:
I agree, but would look the other way here, given that an import job currently takes close to an hour.